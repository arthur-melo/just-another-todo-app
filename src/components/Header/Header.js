import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';

const headerPropTypes = {
  className: PropTypes.string,
};

const Header = ({ className }) => (
  <div className={className}>
    <div className="header__container">
      <nav className="navbar">
        <span className="header__span navbar-brand">
          <FontAwesomeIcon className="d-inline-block" icon={faListAlt} size="lg" alt="List emoji" />
        </span>
        <span className="header__span">Just Another Todo App</span>
      </nav>
    </div>
  </div>
);

Header.propTypes = headerPropTypes;

export default Header;
