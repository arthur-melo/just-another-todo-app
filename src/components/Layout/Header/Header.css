.header__container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background: var(--background-color);
}

.header__span {
  margin: 0 0.2em;
  vertical-align: super;
  font-size: 1.3em;
}
