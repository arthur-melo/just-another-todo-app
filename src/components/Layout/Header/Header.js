import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';
import useTheme from '../../../hooks/useTheme';

const Header = () => {
  const [colors] = useTheme();

  return (
    <div className="header__container">
      <nav className="navbar">
        <span className="header__span navbar-brand">
          <FontAwesomeIcon
            className="d-inline-block"
            icon={faListAlt}
            size="lg"
            alt="List emoji"
            color={colors.primary}
          />
        </span>
        <span className="header__span">Just Another Todo App</span>
      </nav>
    </div>
  );
};

export default Header;
