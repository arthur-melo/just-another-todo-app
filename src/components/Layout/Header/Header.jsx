import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faListAlt } from '@fortawesome/free-solid-svg-icons';

import './Header.css';
import useTheme from '../../../hooks/useTheme';

const Header = () => {
  const [colors] = useTheme();

  return (
    <div className="d-flex justify-content-center align-items-center w-100 h-100 bg-light">
      <nav className="navbar">
        <span className="navbar-brand m-0">
          <FontAwesomeIcon
            className="d-inline-block"
            icon={faListAlt}
            size="lg"
            alt="List emoji"
            color={colors.primary}
          />
        </span>
        <span className="header__text align-top fs-5">
          Just Another Todo App
        </span>
      </nav>
    </div>
  );
};

export default Header;
