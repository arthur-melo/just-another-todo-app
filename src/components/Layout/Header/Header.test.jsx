import renderer from 'react-test-renderer';
import { describe, it, expect } from 'vitest';

import Header from './Header';

describe('Header', () => {
  it('should render the Header component', () => {
    const view = renderer.create(<Header />).toJSON();
    expect(view).toMatchSnapshot();
  });
});
