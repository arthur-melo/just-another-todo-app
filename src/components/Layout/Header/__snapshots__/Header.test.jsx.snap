// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render the Header component 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center align-items-center w-100 h-100 bg-light"
  >
    <nav
      class="navbar"
    >
      <span
        class="navbar-brand m-0"
      >
        <svg
          alt="List emoji"
          aria-hidden="true"
          class="svg-inline--fa fa-rectangle-list fa-lg d-inline-block"
          color=""
          data-icon="rectangle-list"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M64 64C28.7 64 0 92.7 0 128L0 384c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-256c0-35.3-28.7-64-64-64L64 64zm96 256a32 32 0 1 1 -64 0 32 32 0 1 1 64 0zm-32-96a32 32 0 1 1 0-64 32 32 0 1 1 0 64zm120-56l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24zm0 128l144 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-144 0c-13.3 0-24-10.7-24-24s10.7-24 24-24z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="header__text align-top fs-5"
      >
        Just Another Todo App
      </span>
    </nav>
  </div>
</DocumentFragment>
`;
